In [1]:
import csv
f = open("guns.csv", "r")
data = list(csv.reader(f))
print(data[0:5])



[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [3]:
headers = data[0]
data = data[1:len(data)]
print(headers)
print(data[0:5])

['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4']
[['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2'], ['6', '2012', '02', 'Suicide', '0', 'M', '17', 'Native American/Native Alaskan', '100', 'Home', '1']]


In [4]:
years = [item[1] for item in  data]
year_counts = {}
for year in years:
    if year in year_counts:
        year_counts[year] +=1
    else:
        year_counts[year] = 1
print(year_counts)

{'2013': 33636, '2012': 33562, '2014': 33599}


In [6]:
import datetime
dates = [datetime.datetime(year=int(item[1]), month=int(item[2]), day=1) for item in data]
print(dates[0:5])






[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


In [7]:
date_counts = {}
for date in dates:
    if date in date_counts:
        date_counts[date] += 1
    else:
        date_counts[date] = 1
print(date_counts)

{datetime.datetime(2014, 8, 1, 0, 0): 2970, datetime.datetime(2013, 1, 1, 0, 0): 2864, datetime.datetime(2013, 10, 1, 0, 0): 2808, datetime.datetime(2013, 4, 1, 0, 0): 2798, datetime.datetime(2013, 3, 1, 0, 0): 2862, datetime.datetime(2012, 8, 1, 0, 0): 2954, datetime.datetime(2012, 9, 1, 0, 0): 2852, datetime.datetime(2014, 6, 1, 0, 0): 2931, datetime.datetime(2014, 1, 1, 0, 0): 2651, datetime.datetime(2012, 10, 1, 0, 0): 2733, datetime.datetime(2012, 3, 1, 0, 0): 2743, datetime.datetime(2013, 7, 1, 0, 0): 3079, datetime.datetime(2012, 4, 1, 0, 0): 2795, datetime.datetime(2014, 7, 1, 0, 0): 2884, datetime.datetime(2014, 10, 1, 0, 0): 2865, datetime.datetime(2014, 2, 1, 0, 0): 2361, datetime.datetime(2012, 6, 1, 0, 0): 2826, datetime.datetime(2012, 5, 1, 0, 0): 2999, datetime.datetime(2012, 7, 1, 0, 0): 3026, datetime.datetime(2013, 12, 1, 0, 0): 2765, datetime.datetime(2012, 12, 1, 0, 0): 2791, datetime.datetime(2014, 9, 1, 0, 0): 2914, datetime.datetime(2013, 8, 1, 0, 0): 2859, datet

In [9]:
#dictionaries to hold counts
sex_counts = {}
race_counts = {}
#create count dictionaries function
#index is the column in data 
def counts(index, dictionary):
    for item in data:
        if(item[index]) in dictionary:
            dictionary[item[index]] +=1
        else:
            dictionary[item[index]] = 1

counts(5, sex_counts)
counts(7, race_counts)
            
print(sex_counts)
print(race_counts)


{'M': 86348, 'F': 14449}
{'Hispanic': 9022, 'Asian/Pacific Islander': 1325, 'Black': 23296, 'Native American/Native Alaskan': 917, 'White': 66237}


In [10]:
f = open("census.csv", "r")
census = list(csv.reader(f))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [14]:
# dictionary to map race in gun deaths to census value
mapping = {"Asian/Pacific Islander": 15834141, "Black":40250635, "Native American/Native Alaskan":3739506, "Hispanic":44618105, "White":197318956}
race_per_hundredk = {}
# loop through race_counts and calculate deaths per 100000 
# round to 2 decimals
for race,value in race_counts.items():
    value = round((value/mapping[race]) * 100000,2)
    race_per_hundredk[race] = value
print(race_per_hundredk)
    
    



{'Hispanic': 20.22, 'Asian/Pacific Islander': 8.37, 'Black': 57.88, 'Native American/Native Alaskan': 24.52, 'White': 33.57}


In [31]:
intents = [item[3] for item in data]
#print(intents)
races = [item[7] for item in data]
#print(races)
homicide_race_per_hundredk = {}
# loop through every race element from data set
for i,race in enumerate(races):
    if intents[i] == "Homicide":
        if race in homicide_race_per_hundredk:
            homicide_race_per_hundredk[race] += 1
        else:
            homicide_race_per_hundredk[race] = 1
print(homicide_race_per_hundredk)
        

{'Hispanic': 5634, 'Asian/Pacific Islander': 559, 'Black': 19510, 'Native American/Native Alaskan': 326, 'White': 9147}


In [33]:
for race,value in homicide_race_per_hundredk.items():
    value = round((value/mapping[race]) * 100000,2)
    homicide_race_per_hundredk[race] = value
print(homicide_race_per_hundredk)

{'Hispanic': 12.63, 'Asian/Pacific Islander': 3.53, 'Black': 48.47, 'Native American/Native Alaskan': 8.72, 'White': 4.64}


Most people murdered are black with 48.47 out of 100K people. Second hispanic with 12.63.